code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -141,11 +143,32 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 
 	// Update the icon
 	update_icon()
+  
+	var/turf/T = get_turf(src) //yogs
+	var/speedloss = 0
+	var/datum/gas_mixture/env = T.return_air()
+	var/temperature = env.temperature
+	if(temperature <= 150)				// 150K optimal operating parameters
+		net_efective = 100
+	else
+		if(temperature >= 1150)		// at 1000K above 150K the efectivity becomes 0
+			net_efective = 0
+			speedloss = netspeed
+		else
+			var/ratio = netspeed * 0.001			// temp per one unit of speedloss
+			speedloss = round((temperature - 150)/ratio)	// exact speedloss
+			net_efective = 100 - speedloss/netspeed		// percantage speedloss ui use only
+	//yogs end
 
 	if(traffic > 0)
-		traffic -= netspeed
-		if (traffic < 0)  //yogs start
-			traffic = 0   //yogs end
+		var/deltaT = netspeed - speedloss  //yogs start
+		if (traffic < deltaT)
+			deltaT = traffic
+			traffic = 0
+		else
+			traffic -= deltaT
+		if(generates_heat)
+			env.temperature += deltaT * heatoutput / env.heat_capacity()   //yogs end
```

Comment:
this is making the temperature drop to several thousand negative degrees, but I'm too tired atm to actually do the math and tell you what needs changing, I'll do it tomorrow I promise

Reply:
unless the temperature overflows i don't  think there is a way to reach negative kelvins

Reply:
https://docs.google.com/spreadsheets/d/1CQD4fS4ZH7-kx2i__IBuLLH9UUWIQXHVBoRwjEuWj5w/edit?usp=sharing 
I *tried* to do the maths and I think this is accurate

Reply:
the heat cap is usualy about 800 but yes i noticed the mystake thanks for the help